src/pot_process/pot_process.cpp

```diff
@@ -0,0 +1,222 @@
+/// Small utility to check powersoftau output and to compute the evaluation of
+/// Lagrange polynomials at tau.
+
+#include "snarks/groth16/powersoftau_utils.hpp"
+
+#include <boost/program_options.hpp>
+#include <fstream>
+
+using namespace libzeth;
+namespace po = boost::program_options;
+using ppT = libff::default_ec_pp;
+
+// -----------------------------------------------------------------------------
+// cli_options
+// -----------------------------------------------------------------------------
+
+// Usage:
+//     pot-process [<options>] <powersoftau file> <degree>
+//
+// Options:
+//     -h,--help              This message
+//     -v,--verbose           Verbose
+//     --check                Check pot well-formedness and exit
+//     --out <file>           Write the lagrange polynomial values to this file
+//                            ("lagrange-radix2-<n>")
+//     --lagrange-degree <l>  Use degree l instead of n (l < n)
+//     --dummy                Create dummy powersoftau data (for testing only!)
+class cli_options
+{
+public:
+    po::options_description desc;
+    po::options_description all_desc;
+    po::positional_options_description pos;
+
+    std::string argv0;
+    bool help;
+    std::string powersoftau_file;
+    size_t degree;
+    bool verbose;
+    bool check;
+    bool dummy;
+    std::string out;
+    size_t lagrange_degree;
+
+    cli_options();
+    void parse(int argc, char **argv);
+    void usage() const;
+};
+
+cli_options::cli_options()
+    : desc("Options")
+    , all_desc("")
+    , pos()
+    , argv0("pot-process")
+    , help(false)
+    , powersoftau_file()
+    , degree(0)
+    , verbose(false)
+    , check(false)
+    , out()
+    , lagrange_degree(0)
+{
+    desc.add_options()("help,h", "This help")("verbose,v", "Verbose output")(
+        "check", "Check pot well-formedness and exit")(
+        "out,o", po::value<std::string>(), "Output file")(
+        "lagrange-degree", po::value<size_t>(), "Use degree l")(
+        "dummy", "Create dummy powersoftau data (!for testing only)");
+    all_desc.add(desc).add_options()(
+        "powersoftau_file", po::value<std::string>(), "powersoftau file")(
+        "degree", po::value<size_t>(), "degree");
+    pos.add("powersoftau_file", 1).add("degree", 1);
+}
+
+void cli_options::usage() const
+{
+    std::cout << "Usage:" << std::endl
+              << "  " << argv0 << " [<options>] <powersoftau file> <degree>"
+              << std::endl
+              << std::endl
```

Comment:
Replace:
```cpp
              << std::endl
              << std::endl
```
by
```cpp
              << "\n\n"
              << desc << std::endl;
```
To flush a single time.

Reply:
Switched to `"\n\n"`.